src/agent/src/image.rs

```diff
@@ -63,6 +63,27 @@ impl ImageService {
             image_client.config.auth = true;
         }
 
+        #[cfg(feature = "guest-pull")]
+        {
+            let enable_signature_verification = &AGENT_CONFIG.enable_signature_verification;
+            debug!(sl(), "Enable image signature verification: {:?}", enable_signature_verification);
+            image_client.config.security_validate = enable_signature_verification.clone();
+        }
+
+        #[cfg(feature = "guest-pull")]
+        if !&AGENT_CONFIG.image_policy_file.is_empty() {
+            let image_policy_file = &AGENT_CONFIG.image_policy_file;
+            debug!(sl(), "Use imagepolicy file {:?}", image_policy_file);
+            image_client.config.file_paths.policy_path = image_policy_file.clone();
+        }
+
+        #[cfg(feature = "guest-pull")]
+        if !&AGENT_CONFIG.simple_signing_sigstore_config.is_empty() {
+            let simple_signing_sigstore_config = &AGENT_CONFIG.simple_signing_sigstore_config;
+            debug!(sl(), "Using RedHat Simple Signing image signature file {:?}", simple_signing_sigstore_config);
+            image_client.config.file_paths.sigstore_config = simple_signing_sigstore_config.clone();
```

Comment:
Maybe it's an anti-pattern, but did you consider putting all of these sections inside the same `  #[cfg(feature = "guest-pull")] {` block?

Reply:
I thought the same.

Reply:
Yes! Let me merge this